src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -21,520 +21,550 @@
+         }
+         else
+         {
+             if (workingOffset.getZ() <= -field.getWidthMinusZ() && workingOffset.getX() <= -field.getLengthMinusX())
+             {
+                 workingOffset = null;
+                 return false;
+             }
+             else if (workingOffset.getX() <= -field.getLengthMinusX())
+             {
+                 workingOffset = new BlockPos(field.getLengthPlusX(), 0, workingOffset.getZ() - 1);
+             }
+             else
+             {
+                 workingOffset = new BlockPos(workingOffset.getX() - 1, 0, workingOffset.getZ());
+             }
+         }
+         return true;
+     }
+
+     /**
+      * Resets the basic variables of the class.
+      */
+     private void resetVariables()
+     {
+         requestSeeds = true;
+         shouldDumpInventory = true;
+         shouldTryToGetSeed = true;
+     }
+
+     /**
+      * Get's the slot in which the hoe is in.
+      *
+      * @return slot number
+      */
+     private int getHoeSlot()
+     {
+         return InventoryUtils.getFirstSlotContainingTool(getInventory(), Utils.HOE);
+     }
+
+     /**
+      * Executes the planting of the field.
+      *
+      * @return the next state.
+      */
+     @NotNull
+     private AIState plant()
+     {
+         @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+
+         if (buildingFarmer == null || buildingFarmer.getCurrentField() == null)
+         {
+             return AIState.PREPARING;
+         }
+         @Nullable Field field = getOwnBuilding().getCurrentField();
+
+         if (workingOffset != null)
+         {
+             BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
+             if (walkToBlock(position.up()))
+             {
+                 return AIState.FARMER_PLANT;
+             }
+
+             if (shouldPlant(position, field) && !plantCrop(field.getSeed(), position) && !requestSeeds)
+             {
+                 workingOffset = null;
+                 resetVariables();
+                 buildingFarmer.getCurrentField().setNeedsWork(false);
+                 buildingFarmer.getCurrentField().setFieldStage(PLANTED);
+                 return AIState.IDLE;
+             }
+         }
+
+         if (!handleOffset(field))
+         {
+             return terminatePlanting(buildingFarmer, field);
+         }
+
+         setDelay(workingDelay);
+
+         return AIState.FARMER_PLANT;
+     }
+
+     /**
+      * Checks if the ground should be planted.
+      *
+      * @param position the position to check.
+      * @param field    the field close to this position.
+      * @return true if should be hoed.
+      */
+     private boolean shouldPlant(@NotNull BlockPos position, @NotNull Field field)
+     {
+         @Nullable ItemStack itemStack = BlockUtils.getItemStackFromBlockState(world.getBlockState(position.up()));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'itemStack' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)